### ISSUE TITLE
Update to latest RAML 1.0

### ISSUE THREAD

**Issue by kevinduffey** [mulesoft-labs/raml-generator#18]
I see the last update was several months ago to over a year ago. It is not clear if this now supports the latest RAML 1.0 completely or not? Can I parse any RAML 1.0 document, including those with libraries, extensions and more? If not, what will it take to update this project to support RAML 1.0 completely?

**Comment by sichvoge** [comment]
We will come back to this soon. No worries ;) In the meantime, any contribution is very welcome of course. Since that project only parses and handles the generation, you would only need to update the parser afaik. But would need to do some testing of course.

**Comment by kevinduffey** [comment]
So, I see it was updated to RAML 1.0 AST, but I am not quite clear if that means it supports the ability to fully parse RAML 1.0, including annotations, types, includes, etc.. giving the user of this tool the ability to fully support RAML 1.0 or only a subset of it?

**Comment by sichvoge** [comment]
What do you mean by "fully parse RAML 1.0"? It's using the official parser and this project does nothing more than creating all necessary functions for generating something based on your favorite templating framework on a given RAML input. The RAML input is the output of the parser.

**Comment by kevinduffey** [comment]
Well.. I was under the assumption that updates/fixes to the RAML 1.0 JS
engine have been done in the past few months, and the last update shows
RAML 1.0 AST. Maybe I am misunderstanding.. but I assumed for any
additions/fixes to RAML 1.0 JS parser to be incorporated, there would be
more recent updates to this project regarding those. I could totally be
wrong.

What I mean by fully parse RAML 1.0 is that if I feed a fully baked RAML
1.0 set of files, including included files, RAML types, annotations, etc..
can I use this project to get at the entire RAML 1.0 structure... can I set
up templating (e.g. via Handlebars) to access the list of uses.. files,
which if they are RAML types, can I then access the details of each type..
e.g. is it a String or Enum,, the name of it, any min/max properties, etc?
Can I parse the annotation types, and then find them in the RAML file(s)
and make use of them in my template?

In other words, can I essentially use this tool, as is, like I did in the
past with RAML 0.8, to generate any sort of output utilzing the full RAML
1.0 spec (assuming the input files make up a valid RAML 1.0 document)

**Comment by sichvoge** [comment]
In short, yes.
